frontend/src/components/molecules/Forms/RegisterForm.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useContext } from 'react';
+import { validateUser } from 'utils/registrationValidator'
+import { FormControl, Input, Flex, Button, Heading } from '@chakra-ui/react'
+import "./registerForm.css"
+import { registerUser } from 'api/requests';
+import { login } from 'api/auth';
+import { AuthenticateContext } from 'context/authProvider';
+import { errorMonitor } from 'stream';
+import { useHistory } from 'react-router';
+
+
+const RegisterUser: React.FC = () => {
+    const [name, setName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [mobileNumber, setMobileNumber] = useState<number>();
+    const [email, setEmail] = useState('');
+    const [date, setDate] = useState<string>('');
+    const [password, setPassword] = useState<string>('');
+    const [passwordValidation, setPasswordValidation] = useState<string>('');
+    const [username, setUsername] = useState<string>('');
+    
+    const history = useHistory();
+    const [regErr, setError] = useState<undefined | string []>();
+    const { setAuthenticated } = useContext(AuthenticateContext);
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        const UserPayload = { 
+            first_name: name, 
+            last_name: lastName, 
+            password: password,
+            password_validation: passwordValidation,
+            email: email,
+            username: username,
+            mobile_number: mobileNumber,
+            date_of_birth: date,
+        };
+        const errMsg = validateUser(UserPayload);
+        if (errMsg.length) {
+            setError(errMsg);
+            return;
+        }
+
+        try{
+            await registerUser(UserPayload);
+        }
+        catch (error){
+            switch (error.statusCode){
+                case 400:
+                    setError(error.message['detail']);
+                    break;
+                case 409:
+                    setError(error.message['detail']);
+                    break;
+                case 422:
+                default:
+                    setError('Unexpected error, could\'t create user');
+            }
+            return;
+        }
+        try{
+            await login(email, password);
+            setAuthenticated(true);
+        }
+        catch(error){
+            alert(`User created, but could\'t log in, redirecting to start page`);
```

Comment:
Personally i would avoid alert, use chakra toast

Reply:
OK, i will fix this

Reply:
i fixed it